addon/components/abstract-input.js

```diff
@@ -102,26 +102,25 @@ export default Component.extend(PropTypeMixin, {
 
   /**
    * This should be overriden by inherited inputs to convert the value to the appropriate format
-   * @param {Boolean|String} value - value to parse
+   * @param {Boolean|String} data - value to parse
    * @returns {any} parsed value
    */
-  parseValue (value) {
-    return value
+  parseValue (data) {
+    return data.value || _.get(data, 'target.value') || data
```

Comment:
The boolean input overrides this with a specific treatment of `parseValue` that fixes the problem.  This is needed for select input to work correctly, as it returns a straight value, not an object.